test/e2e/common_suite_test.go

```diff
@@ -73,38 +101,69 @@ func (tc *testCase) run() {
 					t.Fatal(err)
 				}
 			}
-
-			if err = client.Resources().Create(ctx, tc.pod); err != nil {
-				t.Fatal(err)
+			if tc.job != nil {
+				if err = client.Resources().Create(ctx, tc.job); err != nil {
+					t.Fatal(err)
+				}
+				if err = wait.For(conditions.New(client.Resources()).JobCompleted(tc.job), wait.WithTimeout(WAIT_JOB_RUNNING_TIMEOUT)); err != nil {
+					t.Log(err)
```

Comment:
IIUC you don't want to `t.Fatal(err)` because you will assess how many pods succeeded vs failed. If it is the case then could you please leave an inline comment here explaining that logic?